src/borg/archiver.py

```diff
@@ -1363,10 +1363,11 @@ def _list_repository(self, args, manifest, write):
         output_data = []
 
         for archive_info in manifest.archives.list_considering(args):
+            archive = Archive(repository, key, manifest, archive_info.name)
```

Comment:
This is problematic, because it is synchronous. So listing 1000 archives now requires 1000 round-trips to the repository. I.e. about a minute for over the Internet.

This needs to be conditional on whether the comment formatters are used. See `format_needs_cache` (`format_keys = {f[1] for f in Formatter().parse(format)}`).